### PR TITLE
chore(usage): update usage collection data format

### PR DESCRIPTION
**Pull request by heiruwu**
Because

- provide more detailed information in usage collection

This commit

- add usage collection execute data and update usage data format
- add redis for caching

### CODE REVIEW

**Comment by donch1989** [comment]
Another thing is that if we don't enable usage function, the datapoints will be pushed into the queue, but nobody pull them. The queue will explode.

File: pkg/service/influx.go [hunk 1]
```diff
@@ -1,7 +1,22 @@
...
-func (s *service) WriteNewDataPoint(p *write.Point) {
-	s.influxDBWriteClient.WritePoint(p)
+	"google.golang.org/protobuf/types/known/structpb"
+
+	"github.com/instill-ai/connector-backend/pkg/utils"
+)
+
+func (s *service) WriteNewDataPoint(ctx context.Context, data utils.UsageMetricData, pipelineMetadata *structpb.Value) {
+
+	s.redisClient.RPush(ctx, fmt.Sprintf("user:%s:execute.execute_time", data.OwnerUID), data.ExecuteTime.Format(time.RFC3339Nano))
```
[the remark this answers is not in the document]
Comment (heiruwu) [reply pkg/service/influx.go:15]: after each report sent, those lists will be cleaned up. but yes it'll be much better if can be done with one list. let me look into it

**Comment by heiruwu** [comment]
> Another thing is that if we don't enable usage function, the datapoints will be pushed into the queue, but nobody pull them. The queue will explode.

will add `usage.enabled` check before write into cache

**Review by donch1989 — APPROVED** [review]
LGTM

**Pull request merged**